packages/metro-file-map/src/HasteFS.js

```diff
@@ -103,3 +150,10 @@ export default class HasteFS {
     return this._files.get(relativePath);
   }
 }
+
+function normalizeSlashes(path: string): string {
+  if (/^\\\\\?\\/.test(path) || /[^\u0000-\u0080]+/.test(path)) {
```

Comment:
I think this needs a comment and/or a function rename - I don't think it's very clear what the intention is. 

It looks like the first test checks for [Win32 file namespaces](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#win32-file-namespaces), but I'm not entirely sure why.

The second test ensures that we return the original path if it contains non-ASCII-128 characters, right? So `'foo\bar'` -> `'foo/bar'` but `foo\bā` -> `foo\bā` - on the face of it that doesn't seem i18n friendly - why the different treatment?

Reply:
My original intention was to make this path work on windows and fail otherwise, that requirement doesn't appear necessary anymore so I'll simply inline the replacement.